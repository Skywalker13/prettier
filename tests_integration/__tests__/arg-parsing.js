"use strict";

const runPrettier = require("../runPrettier");

test("boolean flags do not swallow the next argument", () => {
  const result = runPrettier("cli/arg-parsing", ["--single-quote", "file.js"]);

  expect(result.stdout).toMatchSnapshot();
  expect(result.stderr).toMatchSnapshot();
  expect(result.status).toEqual(0);
});

test("negated options work", () => {
  const result = runPrettier("cli/arg-parsing", ["--no-semi", "file.js"]);

  expect(result.stdout).toMatchSnapshot();
  expect(result.stderr).toMatchSnapshot();
  expect(result.status).toEqual(0);
});

test("unknown options are warned", () => {
  const result = runPrettier("cli/arg-parsing", ["file.js", "--unknown"]);

  expect(result.stdout).toMatchSnapshot();
  expect(result.stderr).toMatchSnapshot();
  expect(result.status).toEqual(0);
});

test("unknown negated options are warned", () => {
  const result = runPrettier("cli/arg-parsing", ["file.js", "--no-unknown"]);

  expect(result.stdout).toMatchSnapshot();
  expect(result.stderr).toMatchSnapshot();
  expect(result.status).toEqual(0);
});
