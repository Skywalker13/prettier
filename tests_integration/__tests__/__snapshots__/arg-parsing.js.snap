// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`boolean flags do not swallow the next argument 1`] = `
"/* eslint-disable */

console.log('could be single quote and without semi');
"
`;

exports[`boolean flags do not swallow the next argument 2`] = `""`;

exports[`negated options work 1`] = `
"/* eslint-disable */

console.log(\\"could be single quote and without semi\\")
"
`;

exports[`negated options work 2`] = `""`;

exports[`unknown negated options are warned 1`] = `
"/* eslint-disable */

console.log(\\"could be single quote and without semi\\");
"
`;

exports[`unknown negated options are warned 2`] = `
"Ignored unknown option: --no-unknown

"
`;

exports[`unknown options are warned 1`] = `
"/* eslint-disable */

console.log(\\"could be single quote and without semi\\");
"
`;

exports[`unknown options are warned 2`] = `
"Ignored unknown option: --unknown

"
`;
